internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -359,3 +362,44 @@ func translateListenerSuppressEnvoyHeaders(suppressEnvoyHeaders *bool, httpIR *i
 		httpIR.SuppressEnvoyHeaders = *suppressEnvoyHeaders
 	}
 }
+
+func translateListenerTLSParameters(tlsParams *egv1a1.TLSSettings, httpIR *ir.HTTPListener) {
+	// Return if this listener isn't a TLS listener. There has to be
+	// at least one certificate defined, which would cause httpIR to
+	// have a TLS structure.
+	if httpIR.TLS == nil {
+		return
+	}
+	// Make sure that the negotiated TLS protocol version is as expected if TLS is used,
+	// regardless of if TLS parameters were used in the ClientTrafficPolicy or not
+	httpIR.TLS.MinVersion = ptr.To(egv1a1.TLSv12)
+	httpIR.TLS.MaxVersion = ptr.To(egv1a1.TLSv13)
+	// If HTTP3 is enabled, the ALPN protocols array should be hardcoded
```

Comment:
cc @tanujd11 , this breaks upgrade or downgrades, right ?
imo this should be an append, not a set

Reply:
I didn't append because if HTTP/3 was enabled, then only HTTP/3 is expected. 

The socket being configured is being explicitly setup as a Quic socket, so I'm not sure that accepting HTTP/1.1 or HTTP/2 would work in this case. It's also what the [original code](https://github.com/envoyproxy/gateway/blob/main/internal/xds/translator/listener.go#L347) did.

Reply:
ah thanks for the clarification